index.test.ts

```diff
@@ -1187,4 +1190,288 @@ describe("Replicate client", () => {
   });
 
   // Continue with tests for other methods
+
+  describe("Stream", () => {
+    function createStream(body: string | NodeJS.ReadableStream) {
+      const streamEndpoint = "https://stream.replicate.com";
+      nock(streamEndpoint)
+        .get(`/fake_stream`)
+        .matchHeader("Accept", "text/event-stream")
+        .reply(200, body);
+
+      return new Stream({ url: `${streamEndpoint}/fake_stream`, fetch });
+    }
+
+    test("consumes a server sent event stream", async () => {
+      const stream = createStream(
+        `
+        event: output
+        id: EVENT_1
+        data: hello world
+
+        event: done
+        id: EVENT_2
+        data: {}
+      `
+          .trim()
+          .replace(/^[ ]+/gm, "")
+      );
+
+      const iterator = stream[Symbol.asyncIterator]();
+
+      expect(await iterator.next()).toEqual({
+        done: false,
+        value: { event: "output", id: "EVENT_1", data: "hello world" },
+      });
+      expect(await iterator.next()).toEqual({
+        done: false,
+        value: { event: "done", id: "EVENT_2", data: "{}" },
+      });
```

Comment:
I think this is a bug, rather than a feature because it means that we always need to check `event.event === "output"` before accessing `event.data`. The `toString` function kind of hides this but it doesn't work with `JSON.stringify()`.

I think actually we want to return the done event from the iterator rather than yielding it. Then the `done` iterator will look like:

```
expect(await iterator.next()).toEqual({ done: true, value: { event: "done", ... } });
```

Reply:
That sounds reasonable to me. Any reason not to make that change to return for done?

Reply:
It's technically a breaking change, but unclear how much of an issue it would cause. There's nothing you can do with it except listen for it and ignore it.